src/lib/background/background.js

```diff
@@ -155,4 +166,24 @@ export class Background {
       showNotification(notifyChangeCount);
     }
   }
+
+  /**
+   * Called whenever scan queue state changes and sends message to other
+   * listeners.
+   *
+   * @param {object} stateData - Scan queue state data.
+   * @param {?Function} sendResponse - Response function.
+   * @private
+   */
+  _handleScanQueueStateChange(stateData, sendResponse) {
+    const message = {
+      action: uiActionsEnum.QUEUE_STATE_CHANGED,
+      data: stateData,
+    };
+    if (sendResponse != null) {
```

Comment:
I think I'd prefer this to be two unidirectional messages, so QUEUE_STATE_REQUEST doesn't have a sendResponse handler, it just triggers a QUEUE_STATE_CHANGED message via `browser.runtime.sendMessage`. That way there's only a single path for QUEUE_STATE_CHANGED, which is a bit simpler.

Reply:
The sendResponse handler is there so that it only response to the requester, not every single piece that listens to QUEUE_STATE_CHANGED. QUEUE_STATE_CHANGED is only ever triggered from this spot, the actual response comes from a QUEUE_STATE_REQUEST.